From 984696a6172e003fc88c48b911e8387f0f435967 Mon Sep 17 00:00:00 2001
From: Zhong Jinghua <zhongjinghua@huawei.com>
Date: Mon, 21 Nov 2022 23:15:32 +0800
Subject: [PATCH 0082/1398] [Huawei] Revert "[Huawei] scsi: fix scsi io
 performance degradation"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187991

----------------------------------------

This reverts commit acfcdaf61b2322088fa94d7b66a00c8b1d67dbc1.

This patch doesn't actually fix hisi_sas_v3 performance issue.

Signed-off-by: Zhong Jinghua <zhongjinghua@huawei.com>
---
 drivers/scsi/scsi_lib.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/scsi/scsi_lib.c b/drivers/scsi/scsi_lib.c
index 92fe9151d492..9352a309fe55 100644
--- a/drivers/scsi/scsi_lib.c
+++ b/drivers/scsi/scsi_lib.c
@@ -52,10 +52,6 @@
 #define  SCSI_INLINE_SG_CNT  2
 #endif
 
-#ifndef test_and_set_bit_relaxed
-#define test_and_set_bit_relaxed test_and_set_bit
-#endif
-
 static struct kmem_cache *scsi_sense_cache;
 static struct kmem_cache *scsi_sense_isadma_cache;
 static DEFINE_MUTEX(scsi_sense_cache_mutex);
@@ -1601,7 +1597,7 @@ static void scsi_mq_done(struct scsi_cmnd *cmd)
 {
 	if (unlikely(blk_should_fake_timeout(cmd->request->q)))
 		return;
-	if (unlikely(test_and_set_bit_relaxed(SCMD_STATE_COMPLETE, &cmd->state)))
+	if (unlikely(test_and_set_bit(SCMD_STATE_COMPLETE, &cmd->state)))
 		return;
 	trace_scsi_dispatch_cmd_done(cmd);
 	blk_mq_complete_request(cmd->request);
-- 
2.17.1

